arctic/chunkstore/chunkstore.py

```diff
@@ -188,19 +203,31 @@ def read(self, symbol, chunk_range=None, columns=None, filter_data=True):
                 }
 
         if chunk_range is not None:
-            spec.update(self.chunker.to_mongo(chunk_range))
+            spec.update(CHUNKER_MAP[sym[CHUNKER]].to_mongo(chunk_range))
 
         segments = []
-        for x in self._collection.find(spec, sort=[(START, pymongo.ASCENDING)],):
-            segments.append(x[DATA])
+        parts = []
+        for x in self._collection.find(spec, sort=[(START, pymongo.ASCENDING), (SEGMENT, pymongo.ASCENDING)],):
+            if x[SEGMENT] > -1:
+                parts.append(x[DATA])
+            else:
+                if parts:
```

Comment:
Ah, I see that segments of chunks are treated specially.  Is it not possible to to treat the segments as just another chunk - this make the code a fair bit more complicated...